unit_tests.sh

```diff
@@ -0,0 +1,9 @@
+#!/bin/sh
+
+# cd to the script directory:
+cd "${0%/*}" || { echo "Couldn't cd to ${0%/*}!"; exit 1; }
+
+export UNIT_TESTS_DIR=./build/out/bin
+
+${UNIT_TESTS_DIR}/bwl_dummy_unit_tests
```

Comment:
ctest is meant to do exactly this, so why don't we just use ctest? Then we don't need to bother with build directory etc.

Reply:
I've never used ctest (nor even cmake) and some help would be very appreciated. 

As you have already approved this changes, I understand that ctest and the integration of unit tests into the CI pipeline can be left for a future PR. 

For the moment, I'll remove this file as it adds little if no value.

Reply:
I have a some experience with ctest - it is integrated cleanly with CMake and it is possible to have it as part of the build. However, I feel that we need a docker for tests.
`build docker` - for building everything.
`test docker` - for testing like `ctest`, `test_flow`, `boardfarm`.
`runner docker` - for running the controller, and maybe also dummy agents - not only for tests, but as a running product.
@arnout - what do yo think?

Reply:
I see little value in having separate docker images. Currently we have a runner and a builder image, but I have no idea why they would be separate images. Well, the builder has a specific entrypoint but it really doesn't *have* to have that.

The docker that does test_flows and boardfarm does have to be a different one, because it's a docker-in-docker image. Note BTW that in the test_flows and boardfarm cases, the docker-in-docker image doesn't run any prplmesh code - it launches a different docker (runner image) for the actual test. So running the unit test code in *that* image makes little sense.

So if we're going to change the docker images, I'd just merge the builder and runner image and run all tests in that one.

Reply:
> I've never used ctest (nor even cmake) and some help would be very appreciated.

ctest is simply a tool that does the equivalent of `make test` in autotools: it discovers the tests (everything marked with add_test in CMakeLists.txt), executes them all, and collects the test results. The simplest use is to do `ctest` in the build directory.

Reply:
bwl_xxx_unit_tests was already marked as a test with add_test() in CMakeLists.txt.

Keep in mind that CTest runs **all** tests and produces a test **summary only**:
```
$ ctest
Test project /home/user1/Projects/prplMesh/build
    Start 1: tlvf_test
1/8 Test #1: tlvf_test ........................   Passed    0.20 sec
    Start 2: encryption_test
2/8 Test #2: encryption_test ..................   Passed    0.11 sec
    Start 3: local_bus_test
3/8 Test #3: local_bus_test ...................   Passed    0.11 sec
    Start 4: message_test
4/8 Test #4: message_test .....................   Passed    0.05 sec
    Start 5: poller_test
5/8 Test #5: poller_test ......................   Passed    0.11 sec
    Start 6: socket_test
6/8 Test #6: socket_test ......................   Passed    1.10 sec
    Start 7: ieee1905_transport_test
7/8 Test #7: ieee1905_transport_test ..........   Passed    0.60 sec
    Start 8: bwl_dummy_unit_tests
8/8 Test #8: bwl_dummy_unit_tests .............   Passed    0.06 sec

100% tests passed, 0 tests failed out of 8

Total Test time (real) =   2.38 sec
```
However, running a test individually produces a **detailed output**:
```
$ ./build/out/bin/tests/bwl_dummy_unit_tests
[==========] Running 6 tests from 1 test suite.
[----------] Global test environment set-up.
[----------] 6 tests from nl80211_client_test
[ RUN      ] nl80211_client_test.get_channel_utilization_should_succeed_with_100_percent
[       OK ] nl80211_client_test.get_channel_utilization_should_succeed_with_100_percent (0 ms)
[ RUN      ] nl80211_client_test.get_channel_utilization_should_succeed_with_50_percent
[       OK ] nl80211_client_test.get_channel_utilization_should_succeed_with_50_percent (0 ms)
[ RUN      ] nl80211_client_test.get_channel_utilization_should_succeed_with_0_percent
[       OK ] nl80211_client_test.get_channel_utilization_should_succeed_with_0_percent (0 ms)
[ RUN      ] nl80211_client_test.get_channel_utilization_should_fail_if_no_channel_in_use
[       OK ] nl80211_client_test.get_channel_utilization_should_fail_if_no_channel_in_use (0 ms)
[ RUN      ] nl80211_client_test.get_channel_utilization_should_fail_if_time_on_is_zero
[       OK ] nl80211_client_test.get_channel_utilization_should_fail_if_time_on_is_zero (0 ms)
[ RUN      ] nl80211_client_test.get_channel_utilization_should_fail_if_time_busy_is_greater_than_time_on
[       OK ] nl80211_client_test.get_channel_utilization_should_fail_if_time_busy_is_greater_than_time_on (0 ms)
[----------] 6 tests from nl80211_client_test (1 ms total)

[----------] Global test environment tear-down
[==========] 6 tests from 1 test suite ran. (1 ms total)
[  PASSED  ] 6 tests.
```

While developing, we'll need to run tests individually so in case any of them fails we get a detailed description of the failure.

Reply:
Update: 
- to get a detailed console output with ctest, use `--verbose` option (useful when run manually)  
- detailed output can be found after running ctest in build/Testing/Temporary/LastTest.log (useful to add it to the CI artifacts)